sympy/physics/mechanics/kane.py

```diff
@@ -330,15 +360,64 @@ def _initialize_constraint_matrices(self, config, vel, acc, linear_solver='LU'):
             self._k_dnh = Matrix()
             self._Ars = Matrix()
 
+        # Initialize nonlinear velocity constraints
+        # The idea is this:
+        # 0 = nonlin_vel.diff(t) is linear in udot, just like vel is linear in u.
+        if nonlin_vel:
+
+            # When calling kanes_equations, another class instance will be
+            # computation of kinetic differential equation matrices will be
+            # skipped as this was computed during the original KanesMethod
+            # object, and the qd_u_map will not be available.
+            if self._qdot_u_map is not None:
+                nonlin_vel = msubs(nonlin_vel, self._qdot_u_map)
+            nonlin_veldt = nonlin_vel.diff(dynamicsymbols._t)
+            self._k_c, _f_c_neg = linear_eq_to_matrix(nonlin_veldt, self._udot[:])
+            self._f_c = _f_c_neg
```

Comment:
You've accidentally removed a minus `-`. I'm curious why your test didn't raise an error.

Reply:
Thanks! I will immediately correct it and push it again.
My guess as to why my test program did not get it: The system are tow points on a horizontal plane. The nonlinear velocity constraint is that their speeds be perpendicular: $u_1 \dot u_2 = 0$ The only forces in the system are the reaction forces needed to ensure this. I guess, if $f_x$ does the job, so does $-f_x$.
Of course, only my guess.